src/export-plan/pdf-gen.ts

```diff
@@ -0,0 +1,328 @@
+import jsPDF from 'jspdf';
+import autoTable from 'jspdf-autotable';
+import {
+  getCollegeFullName,
+  getMajorFullName,
+  getMinorFullName,
+  sortedSemesters,
+  isPlaceholderCourse,
+  getGradFullName,
+} from '../utilities';
+import store from '../store';
+import { addFonts } from './add-fonts';
+import { getCollegeAbbr } from '@/requirements/data/index';
+
+const rowHeight = 18;
+const tableWidth = 516;
+// vertical space between tables for two semesters
+const tableGap = 20;
+const firstTableY = 170;
+
+const rowFontSize = 10.5;
+const headerFontSize = 10.5;
+
+const reqsToFilterOut = ['A&S Credits'];
+
+const bubbleColourMap: Record<RequirementGroupType, (req?: string) => string> = {
+  College: (req?: string) =>
+    req && store.state.userRequirementsMap[req].sourceSpecificName === 'UNI'
+      ? '#1AA9A5'
+      : '#4F7D91',
+  Grad: () => '#1E8481',
+  Major: () => '#1E8481',
+  Minor: () => '#145351',
+};
+
+export const genPDF = (): void => {
+  /* eslint new-cap: "off" */
+  const doc = new jsPDF({ unit: 'pt', format: 'letter' });
+
+  addFonts(doc);
+  doc.setFont('ProximaNova-Regular', 'normal');
+
+  const tableX = (doc.internal.pageSize.width - tableWidth) / 2;
+
+  const img = new Image();
+  img.src = 'src/assets/cp-logo-pdf.png';
+  doc.addImage(img, 48, 30, 84, 23.25);
```

Comment:
Yeah, I just got them from the Figma design. It would be a bit awkward to make them each a constant, since they're all just used once.

Reply:
Would it be possible to make these numbers cleaner (eg. round to the nearest 5) and then refactor? I guess in a future PR

Reply:
even if I do round them to the nearest 5, this is just the position attribute for the image, so nothing else will be at the same coordinates as it. So I'm not sure if it makes sense to make these constants - it might even make the code less readable.

Reply:
Are there other components that are positioned relative to this one? Or are all the major components placed absolutely?

I think the more important side effect of pulling out magic numbers (even to local variables) is to give them some semantic context, not just to refactor multiple instances of the constant. I think having a bunch of magic numbers and absolute positioning in this file makes it more difficult to maintain by other devs (eg. if we change the design, it might be the same LOE to write it mostly from scratch vs finding the right numbers to update).